# Analyse des statistiques de jeu des joueurs de Ligue 1

## Objectifs

Réaliser une analyse exploratoire des données (EDA) sur le dataset des joueurs des 5 grands championnats.  
  


# À propos du dataset

Ce dataset présente les statistiques des joueurs de la saison 2024-2025 dans les cinq principaux championnats européens, issues de FBref. 

## Colonnes

### Informations de base sur le joueur
- **Player** – Nom du joueur  
- **Nation** – Nationalité  
- **Pos** – Poste (FW, MF, DF, GK)  
- **Squad** – Club  
- **Comp** – Championnat  
- **Age** – Âge du joueur  
- **Born** – Année de naissance  

### Temps de jeu et apparitions
- **MP** – Matchs joués  
- **Starts** – Titularisations  
- **Min** – Minutes jouées  
- **90s** – Nombre de matches complets (90 min)  

### Statistiques offensives
- **Gls** – Buts inscrits  
- **Ast** – Passes décisives  
- **G+A** – Buts + passes décisives  
- **xG** – Expected Goals  
- **xAG** – Expected Assists  
- **npxG** – Expected Goals sans pénalty  
- **G-PK** – Buts hors pénalty  

### Statistiques défensives
- **Tkl** – Tacles  
- **TklW** – Tacles réussis  
- **Blocks** – Dégagements (blocs)  
- **Int** – Interceptions  
- **Tkl+Int** – Tacles + interceptions  
- **Clr** – Dégagements  
- **Err** – Erreurs menant à un but  

### Statistiques de passe et créativité
- **PrgP** – Passes progressives  
- **PrgC** – Conduites de balle progressives  
- **KP** – Key Passes (passes menant à un tir)  
- **Cmp%_stats_passing** – Pourcentage de passes réussies  
- **Ast_stats_passing** – Passes décisives (autre colonne dédiée)  
- **xA** – Expected Assists  
- **PPA** – Passes dans la surface de réparation  

### Statistiques des gardiens
- **GA** – Buts encaissés  
- **Saves** – Arrêts  
- **Save%** – Pourcentage d’arrêts  
- **CS** – Clean Sheets  
- **CS%** – Pourcentage de matches sans encaisser  
- **PKA** – Pénaltys subis  
- **PKsv** – Pénaltys arrêtés  

### Possession et contrôle du ballon
- **Touches** – Touches de balle  
- **Carries** – Conduites de balle  
- **PrgR** – Courses progressives (déplacements significatifs vers l’avant)  
- **Mis** – Contrôles manqués  
- **Dis** – Pertes de balle  

### Statistiques diverses
- **CrdY** – Cartons jaunes  
- **CrdR** – Cartons rouges  
- **PKwon** – Pénaltys obtenus  
- **PKcon** – Pénaltys concédés  
- **Recov** – Ballons récupérés  


# 1. Chargement des données

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from itertools import repeat
import plotly.express as px
import plotly.graph_objects as go
from plotly.offline import init_notebook_mode, iplot
from plotly.subplots import make_subplots

In [2]:
football = pd.read_csv('C:/Users/HUGO/OneDrive/portfolio/dataset/test.csv')
football.head()

,Rk,Player,Nation,Pos,Squad,Comp,Age,Born,MP,Starts,...,Att (GK),Thr,Launch%,AvgLen,Opp,Stp,Stp%,#OPA,#OPA/90,AvgDist
0,1,Max Aarons,eng ENG,DF,Bournemouth,eng Premier League,24.0,2000.0,3,1,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,2,Max Aarons,eng ENG,"DF,MF",Valencia,es La Liga,24.0,2000.0,4,1,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,3,Rodrigo Abajas,es ESP,DF,Valencia,es La Liga,21.0,2003.0,1,1,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,4,James Abankwah,ie IRL,"DF,MF",Udinese,it Serie A,20.0,2004.0,6,0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,5,Keyliane Abdallah,fr FRA,FW,Marseille,fr Ligue 1,18.0,2006.0,1,0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [3]:
football.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2854 entries, 0 to 2853
Columns: 165 entries, Rk to AvgDist
dtypes: float64(61), int64(99), object(5)
memory usage: 3.6+ MB


In [4]:
football.describe()

,Rk,Age,Born,MP,Starts,Min,90s,Gls,Ast,G+A,...,Att (GK),Thr,Launch%,AvgLen,Opp,Stp,Stp%,#OPA,#OPA/90,AvgDist
count,2854.000000,2846.000000,2846.000000,2854.000000,2854.000000,2854.000000,2854.000000,2854.000000,2854.000000,2854.000000,...,212.000000,212.00000,212.000000,212.000000,212.000000,212.000000,211.000000,212.000000,212.000000,208.000000
mean,1427.500000,25.018974,1998.637034,19.024177,13.505256,1212.222845,13.468430,1.683952,1.201472,2.885424,...,491.608491,69.45283,34.137736,33.038208,226.556604,14.377358,6.160664,18.768868,1.164528,13.908654
std,824.023159,4.491624,4.497922,11.510177,11.331036,965.819790,10.731516,3.156227,1.946998,4.533767,...,410.277796,57.99046,14.239169,6.066176,187.830723,13.874832,4.075527,18.276921,1.008750,3.734229
min,1.000000,15.000000,1982.000000,1.000000,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,...,1.000000,0.00000,0.000000,6.000000,0.000000,0.000000,0.000000,0.000000,0.000000,2.000000
25%,714.250000,22.000000,1996.000000,9.000000,3.000000,317.250000,3.500000,0.000000,0.000000,0.000000,...,112.250000,15.75000,25.450000,29.475000,55.750000,2.000000,4.000000,3.000000,0.670000,11.975000
50%,1427.500000,25.000000,1999.000000,20.000000,11.000000,1054.000000,11.700000,0.000000,0.000000,1.000000,...,397.500000,55.00000,33.200000,32.450000,175.500000,10.500000,5.600000,14.000000,1.000000,13.700000
75%,2140.750000,28.000000,2002.000000,30.000000,23.000000,1996.750000,22.200000,2.000000,2.000000,4.000000,...,847.250000,120.25000,41.025000,35.900000,408.000000,22.000000,7.900000,30.250000,1.470000,15.525000
max,2854.000000,41.000000,2008.000000,38.000000,38.000000,3420.000000,38.000000,31.000000,18.000000,47.000000,...,1498.000000,197.00000,92.300000,56.300000,710.000000,64.000000,33.300000,89.000000,10.000000,28.000000


# 2. Nettoyage et préparation

- Nation : ne conserver que le code pays à 3 lettres (FRA, ESP, etc.).
- Comp : supprimer le préfixe (ex. "fr Ligue 1" → "Ligue 1").
- Pos : garder le premier poste et mapper vers un nom lisible (GK → Goalkeeper, etc.).
- Filtrage : ne garder que les joueurs ayant joué au moins 17 matches (c'est à dire ayant joué au moins la moitié de la sasion) et sans valeurs manquantes pour Nation, Age ou Pos.
- Création d'une colonne "Pourcentage de pénalty arrété"

In [5]:
football['Nation'] = football['Nation'].str.extract(r'([A-Z]{3})')

football['Comp'] = football['Comp'].str.split(' ', n=1).str[1]

football = football[football['MP'] >= 17]

football = football[football['Nation'].notna()]
football = football[football['Age'].notna()]
football = football[football['Pos'].notna()]

In [6]:
def simplify_position(pos):
    # Si plusieurs postes, prendre le premier
    first_pos = pos.split(',')[0].strip()
    if first_pos == 'GK':
        return 'Goalkeeper'
    elif first_pos == 'DF':
        return 'Defender'
    elif first_pos == 'MF':
        return 'Midfielder'
    elif first_pos == 'FW':
        return 'Forward'
    else:
        return 'Other'

football['Pos'] = football['Pos'].apply(simplify_position)


In [7]:
football['PS%'] = football['PKsv'] / (football['PKA']+football['PKsv']) * 100

In [8]:
football.head()

,Rk,Player,Nation,Pos,Squad,Comp,Age,Born,MP,Starts,...,Thr,Launch%,AvgLen,Opp,Stp,Stp%,#OPA,#OPA/90,AvgDist,PS%
6,7,Himad Abdelli,ALG,Midfielder,Angers,Ligue 1,24.0,1999.0,32,32,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8,9,Ali Abdi,TUN,Defender,Nice,Ligue 1,30.0,1993.0,25,17,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
10,11,Abel,ESP,Defender,Osasuna,La Liga,23.0,2000.0,35,20,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
13,14,Matthis Abline,FRA,Forward,Nantes,Ligue 1,21.0,2003.0,34,33,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
14,15,Abner,BRA,Defender,Lyon,Ligue 1,24.0,2000.0,19,12,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


# 3. Résumé par championnat

On va comparer sur quelques valeurs clées les 5 grands championnats européens

In [9]:
summary = (
    football.groupby('Comp')
      .agg(
         Buts                = ('Gls',       'sum'),
         Match_sans_but_concédé     = ('CS',        'sum'),
         Pénalités_concédées  = ('PKcon',     'sum'),
         Cartons_rouges       = ('CrdR',      'sum'),
         Âge_moyen            = ('Age',       'mean'),
         Tacles_interceptions = ('Tkl+Int',   'mean')
      )
      .round(2)
)

# Mise en évidence des valeurs extrêmes : texte rouge pour le max, vert pour le min
styled = (
    summary.style
           .apply(lambda s: ['color: red'   if v==s.max() else '' for v in s], axis=0)
           .apply(lambda s: ['color: green' if v==s.min() else '' for v in s], axis=0)
)

styled

,Buts,Match_sans_but_concédé,Pénalités_concédées,Cartons_rouges,Âge_moyen,Tacles_interceptions
Comp,,,,,,
Bundesliga,857,132.000000,64,42,26.010000,44.090000
La Liga,913,156.000000,112,70,26.160000,45.830000
Ligue 1,751,120.000000,85,49,25.240000,49.280000
Premier League,1008,151.000000,69,47,25.660000,51.690000
Serie A,827,174.000000,79,61,25.790000,42.690000


- **Premier League**  
  - Total de buts le plus élevé (1082) et moyenne tacles + interceptions la plus forte (33,8).  
  - Cela reflète un championnat à la fois très offensif et intense au pressing, générant beaucoup d’occasions et de récupérations.

- **Serie A**  
  - Nombre de clean sheets le plus élevé (212).  
  - Confirme sa réputation de championnat tactiquement organisé et axé sur la solidité défensive.

- **La Liga**  
  - Plus grand nombre de penalties concédés (130) et de cartons rouges (78).  
  - Traduit un jeu très technique dans la surface, entraînant plus de duels perdus et d’arbitrages stricts.

- **Bundesliga**  
  - Volume de buts et de récupérations (tacles + interceptions) un peu en dessous de la PL, mais toujours élevé, illustrant le style rapide et direct allemand.

- **Ligue 1**  
  - Le plus faible total de buts (85) et le plus petit nombre de clean sheets (146).  
  - 2ᵉ plus de penalties concédés (114) et effectifs les plus jeunes (24,1 ans en moyenne).  
  - Montre un championnat où les matches sont moins prolifiques en buts, où la défense concède plus d’erreurs dans la surface, et où on mise sur des jeunes talents en plein développement.


# 4. Analyse des joueurs de Ligue 1

Dans cette section, nous comparons, de manière équitable, différentes caractéristiques des joueurs de Ligue 1.


In [10]:
ligue1 = football[football["Comp"] == "Ligue 1"]
ligue1.head()

,Rk,Player,Nation,Pos,Squad,Comp,Age,Born,MP,Starts,...,Thr,Launch%,AvgLen,Opp,Stp,Stp%,#OPA,#OPA/90,AvgDist,PS%
6,7,Himad Abdelli,ALG,Midfielder,Angers,Ligue 1,24.0,1999.0,32,32,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8,9,Ali Abdi,TUN,Defender,Nice,Ligue 1,30.0,1993.0,25,17,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
13,14,Matthis Abline,FRA,Forward,Nantes,Ligue 1,21.0,2003.0,34,33,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
14,15,Abner,BRA,Defender,Lyon,Ligue 1,24.0,2000.0,19,12,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
15,16,Zakaria Aboukhlal,MAR,Midfielder,Toulouse,Ligue 1,24.0,2000.0,26,22,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [11]:
ligue1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 276 entries, 6 to 2846
Columns: 166 entries, Rk to PS%
dtypes: float64(62), int64(99), object(5)
memory usage: 360.1+ KB


## Nationalité

In [12]:
from plotly.offline import iplot, init_notebook_mode
init_notebook_mode(connected=True)

fig = px.pie(
    ligue1,
    names='Nation',
    values='MP',
    title="Nombre d'apparitions par pays en Ligue 1"
)

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(
    title_x=0.5,
    width=600,
    height=600,
    showlegend=False
)

fig.show()

In [ ]:
import plotly.express as px
from IPython.display import Image
import plotly.io as pio

fig = px.pie(
    ligue1,
    names='Nation',
    values='MP',
    title="Nombre d'apparitions par pays en Ligue 1"
)

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(
    title_x=0.5,
    width=600,
    height=600,
    showlegend=False
)

# Enregistrer en image statique
pio.write_image(fig, 'ligue1_pie.png')  # Nécessite kaleido installé
Image('ligue1_pie.png')

In [ ]:
fig = px.pie(
    ligue1,
    names='Nation',
    values='MP',
    title="Nombre d'apparitions par pays en Ligue 1"
)

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(
    title_x=0.5,
    width=600,
    height=600,
    showlegend=False
)

iplot(fig)

In [ ]:
fig = px.sunburst(
    ligue1,
    path=['Pos', 'Nation'],
    values='MP',
    title="Répartition des positions par pays"
)

fig.update_layout(
    title_x=0.5,
    width=600,
    height=600,
    showlegend=False
)

iplot(fig)

**Résumé des nations les plus représentées en Ligue 1 :**  
- **Global** : 1ᵉʳ France, 2ᵉᵐᵉ Côte d’Ivoire, 3ᵉᵐᵉ Brésil  
- **Gardiens** : 1ᵉʳ France, 2ᵉᵐᵉ Argentine, 3ᵉᵐᵉ Pologne  
- **Défenseurs** : 1ᵉʳ France, 2ᵉᵐᵉ Brésil, 3ᵉᵐᵉ Pays-Bas  
- **Milieux** : 1ᵉʳ France, 2ᵉᵐᵉ Algérie, 3ᵉᵐᵉ Maroc  
- **Attaquants** : 1ᵉʳ France, 2ᵉᵐᵉ Sénégal, 3ᵉᵐᵉ Belgique  

La forte présence des joueurs français à tous les postes témoigne de la qualité de leur formation, tandis que les filières ivoirienne, sénégalaise et brésilienne complètent et enrichissent le championnat.  


## Age

In [ ]:
age_moy = ligue1['Age'].mean()

# 1. Répartition des âges par poste
fig1 = px.violin(
    ligue1,
    x='Pos',
    y='Age',
    box=True,
    title="Répartition des âges par poste"
)
fig1.add_hline(
    y=age_moy,
    line_dash="dot",
    annotation_text=f"Âge moyen ({age_moy:.1f} ans)",
    annotation_position="bottom right"
)
iplot(fig1)

# 2. Répartition des âges par club
fig2 = px.box(
    ligue1,
    y='Squad',
    x='Age',
    title="Répartition des âges par club"
)
fig2.add_vline(
    x=age_moy,
    line_dash="dot",
    annotation_text=f"Âge moyen ({age_moy:.1f} ans)",
    annotation_position="top left"
)
iplot(fig2)


Les gardiens et les défenseurs sont relativement plus âgés que leurs collègues milieux de terrain et attaquants. La ligne défensive (y compris les gardiens) est largement perçue comme un secteur où l’on privilégie davantage le sang-froid et l’expérience qu’une vitesse fulgurante.
**Résumé :**  
- L’effectif le plus jeune est celui de Strasbourg  
- Brest possède le groupe de joueurs le plus âgé  
- Les âges minimum, médian et maximum sont de 16, 25,2 et 40 ans  


# 5. Analyse par poste

Dans cette prtie de l'analyse on va analyser les 5 meilleurs joueurs à leur poste en ligue 1 sur des données différentes.

## Gardiens

In [ ]:
gk = ligue1[(ligue1['Pos']=='Goalkeeper') & (ligue1['MP']>=17)]

stats = ['Saves', 'Save%', 'CS', 'CS%', 'PKsv', 'PS%']
titles = [
    'Arrêts',
    '% d\'arrêts',
    'Clean sheets',
    '% de clean sheets',
    'Arrêts de pénalty',
    '% d\'arrêts de pénalty'
]

fig = make_subplots(
    rows=3, cols=2,
    subplot_titles=titles,
    horizontal_spacing=0.12,
    vertical_spacing=0.15
)

for i, stat in enumerate(stats):
    top5 = gk.nlargest(5, stat)
    row, col = i//2 + 1, i%2 + 1
    colors = [
        'rgba(46,204,113,0.8)' if v==top5[stat].max() else 'rgba(189,195,199,0.8)'
        for v in top5[stat]
    ]
    fig.add_trace(
        go.Bar(
            y=top5['Player'],
            x=top5[stat],
            orientation='h',
            hovertext=top5['Squad'],
            marker=dict(
                color=colors,
                line=dict(color='gray', width=1.5)
            ),
            opacity=0.9
        ),
        row=row,
        col=col
    )

fig.update_layout(
    title_text="Top 5 des gardiens – statistiques clés",
    title_x=0.5,
    titlefont=dict(size=20),
    showlegend=False,
    width=1300,
    height=900,
    template='ggplot2',
    paper_bgcolor='lightgray'
)

iplot(fig)


In [ ]:
defense = ligue1[(ligue1['Pos']=='Defender') & (ligue1['MP']>=17)]

stats = ['Tkl', 'TklW', 'Int', 'Blocks_stats_defense', 'Tkl+Int', 'Clr']
titles = ['Tacles', 'Tacles réussis', 'Interceptions', 'Blocs', 'Tacles + Interceptions', 'Dégagements']

fig = make_subplots(rows=3, cols=2, subplot_titles=titles,
                    horizontal_spacing=0.12, vertical_spacing=0.15)

for i, stat in enumerate(stats):
    top5 = defense.nlargest(5, stat)
    r, c = divmod(i, 2)
    colors = ['rgba(46,204,113,0.8)' if v==top5[stat].max() else 'rgba(189,195,199,0.8)' 
              for v in top5[stat]]
    fig.add_trace(
        go.Bar(
            y=top5['Player'], x=top5[stat], orientation='h',
            hovertext=top5['Squad'],
            marker=dict(color=colors, line=dict(color='gray', width=1.5)),
            opacity=0.9
        ),
        row=r+1, col=c+1
    )

fig.update_layout(
    title_text="Top 5 des défenseurs – statistiques clés",
    title_x=0.5, showlegend=False,
    width=1300, height=900, template='ggplot2',
    paper_bgcolor='lightgray'
)

iplot(fig)

In [ ]:
midfielders = ligue1[(ligue1['Pos']=='Midfielder') & (ligue1['MP']>=17)]

stats = [
    'PrgP', 'PrgC', 'KP', 'Cmp',
    'Cmp%', 'Ast_stats_passing',
    'Touches', 'Carries'
]
titles = [
    'Passes progressives',
    'Courses progressives',
    'Passes clés',
    'Passes réussies (nombre)',
    '% de passes réussies',
    'Passes décisives',
    'Touches de balle',
    'Conduites de balle'
]

fig = make_subplots(
    rows=4, cols=2,
    subplot_titles=titles,
    horizontal_spacing=0.12,
    vertical_spacing=0.12
)

for i, stat in enumerate(stats):
    top5 = midfielders.nlargest(5, stat)
    r, c = i//2 + 1, i%2 + 1
    colors = [
        'rgba(46,204,113,0.8)' if v==top5[stat].max() else 'rgba(189,195,199,0.8)'
        for v in top5[stat]
    ]
    fig.add_trace(
        go.Bar(
            y=top5['Player'],
            x=top5[stat],
            orientation='h',
            hovertext=top5['Squad'],
            marker=dict(color=colors, line=dict(color='gray', width=1.5)),
            opacity=0.9
        ),
        row=r, col=c
    )

fig.update_layout(
    title_text="Top 5 des milieux de terrain – statistiques clés",
    title_x=0.5,
    titlefont=dict(size=20),
    showlegend=False,
    width=1300,
    height=1400,
    template='ggplot2',
    paper_bgcolor='lightgray'
)

iplot(fig)

In [ ]:
forwards = ligue1[(ligue1['Pos']=='Forward') & (ligue1['MP']>=17)]

stats = ['Gls', 'Ast', 'PrgP', 'PrgC', 'KP', 'PrgR']
titles_fr = [
    'Buts marqués',
    'Passes décisives',
    'Passes progressives',
    'Courses progressives',
    'Passes clés',
    'Courses progressives avec ballon'
]

fig = make_subplots(
    rows=3, cols=2,
    subplot_titles=titles_fr + [''],
    horizontal_spacing=0.12,
    vertical_spacing=0.12
)

for i, stat in enumerate(stats):
    top5 = forwards.nlargest(5, stat)
    r, c = i//2 + 1, i%2 + 1
    colors = [
        'rgba(46,204,113,0.8)' if v==top5[stat].max() else 'rgba(189,195,199,0.8)'
        for v in top5[stat]
    ]
    fig.add_trace(
        go.Bar(
            y=top5['Player'],
            x=top5[stat],
            orientation='h',
            hovertext=top5['Squad'],
            marker=dict(color=colors, line=dict(color='gray', width=1.5)),
            opacity=0.9
        ),
        row=r, col=c
    )

fig.update_layout(
    title_text="Top 5 des attaquants – statistiques clés",
    title_x=0.5,
    titlefont=dict(size=20),
    showlegend=False,
    width=1300,
    height=1400,
    template='ggplot2',
    paper_bgcolor='lightgray'
)

iplot(fig)

In [ ]:
agg = ligue1.groupby(['Pos', 'Squad'])['Gls'].sum().reset_index()

postes = ['Goalkeeper','Defender','Midfielder','Forward']
clubs = agg['Squad'].unique().tolist()
labels = postes + clubs
index = {l: i for i, l in enumerate(labels)}

couleurs_noeuds = ['#3498db','#e74c3c','#f39c12','#2ecc71'] + ['#95a5a6']*len(clubs)
couleurs_liens = {
    'Goalkeeper':'rgba(52,152,219,0.4)',
    'Defender':'rgba(231,76,60,0.4)',
    'Midfielder':'rgba(243,156,18,0.4)',
    'Forward':'rgba(46,204,113,0.4)'
}


fig = go.Figure(go.Sankey(
    node=dict(label=labels, color=couleurs_noeuds),
    link=dict(
        source=agg['Pos'].map(index),
        target=agg['Squad'].map(index),
        value=agg['Gls'],
        color=agg['Pos'].map(couleurs_liens)
    )
))

fig.update_layout(
    title_text="Flux de buts : poste → club",
    title_x=0.5
)

iplot(fig)

- Les attaquants inscrivent le plus de buts (rien d’étonnant).  
- Le PSG domine le classement des buts marqués.  
- Nice enregistre le plus grand nombre de buts de défenseurs.  
- Montpellier est le moins prolifique, avec le total de buts le plus faible.


# 6. Qui mérite le Ballon d’Or : Ousmane Dembélé ou Lamine Yamal ?

In [ ]:
players = ['Lamine Yamal', 'Ousmane Dembélé']
metrics = ['Gls','Ast','G+A','Sh/90','Succ%','KP','Cmp%','Touches','Carries']

df_comp = football[football.Player.isin(players)].set_index('Player')[metrics]
df_norm = df_comp / df_comp.max()

N = len(metrics)
angles = np.linspace(0, 2*np.pi, N, endpoint=False)
angles = np.concatenate([angles, [angles[0]]])

fig, ax = plt.subplots(subplot_kw={'polar':True}, figsize=(6,6))
for player in players:
    vals = np.append(df_norm.loc[player].values, df_norm.loc[player].values[0])
    ax.plot(angles, vals, label=player)
    ax.fill(angles, vals, alpha=0.25)

ax.set_thetagrids(np.degrees(angles[:-1]), metrics)
ax.set_ylim(0,1)
ax.set_title(' Yamal vs Dembélé')
ax.legend(loc='upper left', bbox_to_anchor=(1.05, 1))
plt.tight_layout()
plt.show()

En comparant les statatistiques de Yamal et Dembélé dans leur championnat respectif on se rend compte qu'ils ont deux profils très différents. Yamal est très actif au toucher de balle et en costruction de jeux, Dembélé plus précis en passe, tir et but marqué. 

Il est donc difficile de choisir le ballon d'or que sur ces données. Il fauda prendre en compte aussi celle de la ligue des champions.